it/multipart/src/test/java/com/linecorp/armeria/common/multipart/MultipartCollector.java

```diff
@@ -0,0 +1,287 @@
+                }
+            }
+        });
+        return future;
+    }
+
+    public CompletableFuture<CustomAggregatedMultipart> future() {
+        return future;
+    }
+
+    @Override
+    public void onSubscribe(Subscription s) {
+        multipartSubscription = s;
+        eventLoop = RequestContext.mapCurrent(RequestContext::eventLoop,
+                                              CommonPools.workerGroup()::next);
+        multipartSubscription.request(1);
+    }
+
+    @Override
+    public void onNext(BodyPart bodyPart) {
+        inProgressCount++;
+        if (bodyPart.filename() == null) {
+            final CompletableFuture<AggregatedBodyPart> future = new CompletableFuture<>();
+            // Nested subscription need passing event-loop. Because the onNext is not context-aware.
+            // Avoid it using common worker pool instead of event-loop from RequestContext.
+            bodyPart.content().subscribe(
+                    new HttpObjectAggregator<AggregatedBodyPart>(future, null) {
+                        @Override
+                        protected void onHeaders(HttpHeaders headers) {
+                        }
+
+                        @Override
+                        protected AggregatedBodyPart onSuccess(HttpData content) {
+                            return AggregatedBodyPart.of(bodyPart.headers(), content);
+                        }
+
+                        @Override
+                        protected void onFailure() {
+                        }
+                    }, eventLoop);
+            future.whenCompleteAsync((aggregatedBodyPart, throwable) -> {
+                inProgressCount--;
+                if (throwable != null) {
+                    multipartSubscription.cancel();
+                    future.completeExceptionally(throwable);
+                    return;
+                }
+                @Nullable
+                final String name = aggregatedBodyPart.name();
+                if (name != null) {
+                    @Nullable
+                    final MediaType mediaType = aggregatedBodyPart.contentType();
+                    final Charset charset = mediaType == null ? StandardCharsets.US_ASCII
+                                                              : mediaType.charset(StandardCharsets.US_ASCII);
+                    queryParamsBuilder.add(name, aggregatedBodyPart.content(charset));
+                }
+                multipartSubscription.request(1);
+                if (canComplete) {
+                    doComplete();
+                }
+            }, eventLoop);
+            return;
+        }
+        try {
+            final Path path = Files.createTempFile("armeria", "tmp");
+            files.compute(bodyPart.name(), (s, paths) -> {
+                if (paths == null) {
+                    paths = new ArrayList<>();
+                }
+                paths.add(path);
+                return paths;
+            });
+            eventLoop.execute(() -> {
```

Comment:
Due to `onNext` may not have `RequestContext`. If we just call `subscribe` in  `toPath`, it will use `CommonPools.workerGroup()::next` in `defaultSubscriberExecutor`. So force it use context-awared eventoop.